### ISSUE TITLE
Open source code for the rentry site?

### ISSUE THREAD

**Issue by dessalines** [radude/rentry#1]
I'd be interested in seeing the code for the rentry site, esp with regards to knowing which javascript markdown engine is in use, and such.

**Comment by dessalines** [comment]
Also to potentially add features like secret / encoded pastes and such.

**Comment by radude** [comment]
The code is crap. I will opensource it once it is cleaned up a bit. There isn't any javascript markdown engine though, what I use is [Python-Markdown](https://github.com/Python-Markdown/markdown) with some extensions and [codemirror](https://github.com/codemirror/CodeMirror) to highlight text in the editor.

As for encrypted pastes it is on the todo list.

**Comment by LeBaux** [comment]
While on the topic of the website, maybe consider adding dark mode (maybe even by default?). CLI could have a switch -d or something. I understand it is cosmetic and unnecessary, just throwing it out there.

**Comment by radude** [comment]
>Every pastebin thingy is adding unnecessary crap like [...] comments

Whoops, I was planning on adding (disableable (anonymous)) comments. I'm slow tho, so you don't have to fear the bloat yet.

**Comment by LeBaux** [comment]
I understand people want to make their (pet) projects better. But sometimes, UNIX approach is the best. Do one thing, do it the best way possible.

If I want any sort of collab on MD file, there is already hackmd.io. It is pretty sweet. Or you know, GitHub.

Look at the OG pastebin, I would say 9/10 notes are anon pastes, one time, bye-bye. Be the bigger man, rise above, be happy with what you made.

Could I use an account for managing pastes? Yeah. Do I NEED it? Not at all.

Also, you removed WTF? I have to admit, now it nears perfection.

**Comment by freqmand** [comment]
@radude i would appreciate if you could​ share the web source code by email. 
email : freqmand@protonmail.com

**Comment by KarelWintersky** [comment]
is a sourcecode still closed? Four years...

**Comment by freqmand** [comment]
I didn't receive the code by the way :)

**Comment by KarelWintersky** [comment]
@radude ?

**Comment by Mikle-Bond** [comment]
@radude, maybe you could create a repo with `main` branch pointing at initial commit, and the actual code hidden in `dev` branch. So that it is made publicly available only when you feel happy about it. If the code works, it can be "uncrapped", and there are people willing to do so)

(unless you keep all your passwords tightly packed into the codebase)

**Comment by dejl** [comment]
It would be more productive to just make a clone lol

**Comment by 0aoq** [comment]
Some time ago I was looking for a self-hostable Markdown pastebin and found Rentry, but unfortunately Rentry cannot be hosted yourself. That was a bit of a disappointment, so I thought about making one myself but I just decided it would be too much work for something so niche.

So anyways, I did it over the past day and decided I'd share it here in case any of you are still looking for an open-source Markdown pastebin. It's not perfect, but it does the job. Here it is: https://codeberg.org/hkau/entry (A compatibility table is available in the README)

The UI is (almost) a complete re-creation, with some minor differences because I thought they looked better. It is written in TypeScript for use with the [Bun](https://bun.sh) runtime, this means the server only really runs on Linux/MacOS, but it makes the server much more lightweight than if it was written for Node.

> It would be more productive to just make a clone from scratch, rather than expecting rentry to become FOSS lol

Yeah, it's been four years since this issue was opened and there hasn't really been any developments, so I hope this is useful for someone. It's written in TypeScript because I'm not very good with Python (which is what Rentry uses), and it was just easier

It's built with:
- TypeScript (Bun)
- CodeMirror 6 (for editor)
- Marked (for Markdown rendering)
- Preact (very small JSX renderer)

**Comment by mousemoder** [comment]
To my understanding, Radude no longer maintains Rentry and has passed the project off to someone else.
The current site owner does not wish to open the source code, per the answer provided to this same exact question via Retrospring.
https://retrospring.net/@rentry_official/a/112525352553330212

**Comment by KarelWintersky** [comment]
> and after seeing just how completely overrun new paste sites with little to no moderation become (we've seen some that are so bad that it's a wonder the authorities haven't smashed it down and thrown them in jail by now), it'd almost feel irresponsible to facilitate.

it looks ridiculous. So, as if the content posted on other similar projects is to blame for the fact that the rentry does not go open source.

**Comment by hkauso** [comment]
> it looks ridiculous. So, as if the content posted on other similar projects is to blame for the fact that the rentry does not go open source.

Yeah, it's definitely an interesting comment. I'd say moderation issues with newer "markdown services" really *shouldn't* impact the ability of Rentry to go open source, but at the same time I'd say it's best to respect the new owner's wishes to keep the source closed.

However, if the new owner were to reconsider, I'd say that the main Rentry site could benefit greatly from going open source. I'm sure there's likely a group of individuals who would love to help contribute to Rentry and improve it from its current state.

It would be interesting to see some examples of new paste sites that become "overrun".
